frontends/ol-components/src/components/LearningResourceExpanded/LearningResourceExpandedV2.tsx

```diff
@@ -413,55 +509,112 @@ const CallToActionSection = ({
+          data-ph-resource-type={resource.resource_type}
+          data-ph-resource-id={resource.id}
+          endIcon={<RiExternalLinkLine />}
+          href={getCallToActionUrl(resource) || ""}
+        >
+          {cta}
+        </StyledLink>
+        <PlatformContainer>
+          {platformImage ? (
+            <Platform>
+              <OnPlatform>on</OnPlatform>
+              <StyledPlatformLogo platformCode={platformCode} height={26} />
+            </Platform>
+          ) : null}
+        </PlatformContainer>
+        <ButtonContainer>
           {user?.is_learning_path_editor && (
-            <CardActionButton
-              filled={inLearningPath}
-              aria-label="Add to Learning Path"
+            <CallToActionButton
+              filled={inLearningPath ? 1 : 0}
+              startIcon={<RiMenuAddLine />}
+              aria-label={addToLearningPathLabel}
               onClick={(event) =>
                 onAddToLearningPathClick
                   ? onAddToLearningPathClick(event, resource.id)
                   : null
               }
             >
-              <RiMenuAddLine aria-hidden />
-            </CardActionButton>
+              {addToLearningPathLabel}
+            </CallToActionButton>
           )}
-          <CardActionButton
-            filled={inUserList}
-            aria-label={`Bookmark ${getReadableResourceType(resource.resource_type)}`}
+          <CallToActionButton
+            filled={inUserList ? 1 : 0}
+            startIcon={inUserList ? <RiBookmarkFill /> : <RiBookmarkLine />}
+            aria-label={bookmarkLabel}
             onClick={
               onAddToUserListClick
                 ? (event) => onAddToUserListClick?.(event, resource.id)
                 : undefined
             }
           >
-            <RiBookmarkLine aria-hidden />
-          </CardActionButton>
-        </ListButtonContainer>
-      </PlatformContainer>
+            {bookmarkLabel}
+          </CallToActionButton>
+          <CallToActionButton
+            filled={shareExpanded ? 1 : 0}
+            startIcon={<RiShareLine />}
+            aria-label={shareLabel}
+            onClick={() => setShareExpanded(!shareExpanded)}
+          >
+            {shareLabel}
+          </CallToActionButton>
+        </ButtonContainer>
+        {shareExpanded && shareUrl && (
+          <ShareContainer data-testid="drawer-share">
+            <ShareLabel>Share a link to this Resource</ShareLabel>
+            <ShareInput value={shareUrl} />
+            <ShareButtonContainer>
+              <ShareLink
+                href={`${facebookShareBaseUrl}?u=${encodeURIComponent(shareUrl)}`}
+                target="_blank"
+              >
+                <RiFacebookFill size={socialIconSize} />
+              </ShareLink>
+              <ShareLink
+                href={`${twitterShareBaseUrl}?text=${encodeURIComponent(resource.title)}&url=${encodeURIComponent(shareUrl)}`}
```

Comment:
@sovsey This is actually something I meant to ask about. I put `text` in here because otherwise the tweet is just the URL in the text field, but we probably want something other than the resource title:

![image](https://github.com/user-attachments/assets/e90d4928-6910-4f22-895d-eb54b21f56aa)

It's a bit different to the interface on other platforms like LinkedIn which automatically detect that you're sharing a link and exclude it from the text content of the post:

![image](https://github.com/user-attachments/assets/400a30e1-51d1-4f39-8888-f3783a2a32a8)

So, for Twitter at least, we probably want to frontload some text as that's more traditionally how a tweet is written anyway. However, when you actually go to post a tweet the URL is obfuscated and replaced with the typcial image based social media preview. So, we could leave it as-is but I feel like most people add some text there like "Check out this post on MIT Learn!"

Reply:
Let's have just the title and URL (as you have now) for Twitter/X. Was browsing a few new sites, and that seems to be typical at least.